modules/core/src/test/java/org/apache/ignite/internal/util/HostAndPortRangeTest.java

```diff
@@ -0,0 +1,156 @@
+        HostAndPortRange actual = HostAndPortRange.parse(addrStr, dfltPortFrom, dfltPortTo, errMsgPrefix);
+        HostAndPortRange expected = new HostAndPortRange("127.0.0.1", 8080, 8080);
+        assertEquals(expected, actual);
+    }
+
+    /**
+     * ests correct input address with IPv4 host and no port.
+     * @throws IgniteCheckedException on incorrect host/port
+     */
+    @Test
+    public void testParseIPv4NoPort() throws IgniteCheckedException {
+        String addrStr = "127.0.0.1";
+        String errMsgPrefix = "";
+        int dfltPortFrom = 18360;
+        int dfltPortTo = 18362;
+        HostAndPortRange actual = HostAndPortRange.parse(addrStr, dfltPortFrom, dfltPortTo, errMsgPrefix);
+        HostAndPortRange expected = new HostAndPortRange("127.0.0.1", 18360, 18362);
+        assertEquals(expected, actual);
+    }
+
+    /**
+     * tests correct input address with IPv6 host and port range.
+     * @throws IgniteCheckedException on incorrect host/port
+     */
+    @Test
+    public void testParseIPv6WithPortRange() throws IgniteCheckedException {
+        String addrStr = "[::1]:8080..8090";
+        String errMsgPrefix = "";
+        int dfltPortFrom = 18360;
+        int dfltPortTo = 18362;
+        HostAndPortRange actual = HostAndPortRange.parse(addrStr, dfltPortFrom, dfltPortTo, errMsgPrefix);
+        HostAndPortRange expected = new HostAndPortRange("::1", 8080, 8090);
+        assertEquals(expected, actual);
+    }
+
+    /**
+     * tests correct input address with IPv6 host and single port.
+     * @throws IgniteCheckedException on incorrect host/port
+     */
+    @Test
+    public void testParseIPv6WithSinglePort() throws IgniteCheckedException {
+        String addrStr = "[3ffe:2a00:100:7031::]:8080";
+        String errMsgPrefix = "";
+        int dfltPortFrom = 18360;
+        int dfltPortTo = 18362;
+        HostAndPortRange actual = HostAndPortRange.parse(addrStr, dfltPortFrom, dfltPortTo, errMsgPrefix);
+        HostAndPortRange expected = new HostAndPortRange("3ffe:2a00:100:7031::", 8080, 8080);
+        assertEquals(expected, actual);
+    }
+
+    /**
+     * tests correct input address with IPv6 host and no port.
+     * @throws IgniteCheckedException on incorrect host/port
+     */
+    @Test
+    public void testParseIPv6NoPort() throws IgniteCheckedException {
+        String addrStr = "[::FFFF:129.144.52.38]";
+        String errMsgPrefix = "";
+        int dfltPortFrom = 18360;
+        int dfltPortTo = 18362;
+        HostAndPortRange actual = HostAndPortRange.parse(addrStr, dfltPortFrom, dfltPortTo, errMsgPrefix);
+        HostAndPortRange expected = new HostAndPortRange("::FFFF:129.144.52.38", 18360, 18362);
+        assertEquals(expected, actual);
+    }
+
+    /**
+     * tests incorrect input address with IPv6 host (no brackets) and port.
+     * @throws IgniteCheckedException on incorrect host/port
+     */
+    @Test(expected = IgniteCheckedException.class)
```

Comment:
Please add a check for the exception message too, not only the class.

Reply:
Done